client/components/dialog/dialog-base.jsx

```diff
@@ -63,7 +63,7 @@ class DialogBase extends Component {
 			clickHandler = this._onButtonClick.bind( this, button );
 
 		return (
-			<button key={ button.action } className={ classes } onClick={ clickHandler } disabled={ !! button.disabled }>
+			<button key={ button.action } className={ classes } data-e2e-button={ button.label } onClick={ clickHandler } disabled={ !! button.disabled }>
```

Comment:
Is button.label a translated string? If so it would be better to use an attribute that isn't translated